src/app/zap-templates/zcl/data-model/chip/test-cluster.xml

```diff
@@ -17,11 +17,50 @@ limitations under the License.
 <configurator>
   <domain name="CHIP"/>
 
-   <struct name="TestListStructOctet">
-     <item name="fabricIndex" type="INT64U"/>
-     <item name="operationalCert" type="OCTET_STRING" length="32"/>
+  <struct name="TestListStructOctet">
+    <item name="fabricIndex" type="INT64U"/>
+    <item name="operationalCert" type="OCTET_STRING" length="32"/>
+  </struct>
+
+  <enum name="SimpleEnum" type="ENUM8">
+    <item name="Unspecified" value="0x00"/>
+    <item name="ValueA" value="0x01"/>
+    <item name="ValueB" value="0x02"/>
+    <item name="ValueC" value="0x03"/>
+  </enum>
+  
+  <struct name="SimpleStruct">
+    <item name="a" type="UINT8U" optional="false"/>
+    <item name="b" type="BOOLEAN" optional="false"/>
+    <item name="c" define="SIMPLE_ENUM" type="ENUM8" optional="false"/>
+    <item name="d" type="OCTET_STRING" optional="false"/>
+    <item name="e" type="CHAR_STRING" optional="false"/>
+  </struct>
+
+   <struct name="NestedStruct">
+    <item name="a" type="UINT8U" optional="false"/>
+    <item name="b" type="BOOLEAN" optional="false"/>
+     <!--
+     <item name="c" type="SimpleStruct" optional="false"/>
+     -->
    </struct>
 
+   <struct name="NestedStructList">
+    <item name="a" type="UINT8U" optional="false"/>
+    <item name="b" type="BOOLEAN" optional="false"/>
+     <!--
+    <item name="c" type="SimpleStruct" optional="false"/>
+     -->
+    <item name="d" type="SimpleStruct" array="true" optional="false"/>
```

Comment:
there array are not right in xml, @mrjerryjohns  you should use entryType and type combination for right array representation for zap xml.
    <item name="d" type="ARRAY" entryType="SimpleStruct" optional="false"/>
    <item name="e" type="ARRAY" entryType="UINT32U" optional="false"/>
    <item name="f" type="ARRAY" entryType="OCTET_STRING" optional="false"/>
    <item name="f" type="ARRAY" entryType="UINT8U" optional="false"/>

Reply:
Not sure what the fix here should be.

Could you fix these in your PR for codegen?